tests/test_money.py

```diff
@@ -153,3 +153,47 @@ def test_sub_negative():
     for bill in bills:
         total -= bill
     assert total == Money(-33, "EUR")
+
+
+@pytest.mark.parametrize(
+    "decimal_places_display, decimal_places",
+    [
+        [None, None],
+        [0, 0],
+        [1, 0],
+        [4, 0],
+        [0, 1],
+        [1, 1],
+        [4, 1],
+        [0, 4],
+        [1, 4],
+        [4, 4],
+        [None, 4],
+        [None, 1],
+        [None, 0],
+        [4, None],
+        [1, None],
+        [0, None],
+    ],
+)
+def test_proper_copy_of_attributes(decimal_places_display, decimal_places):
+    one = Money(1, "EUR", decimal_places_display=decimal_places_display)
+
+    assert one._decimal_places_display is decimal_places_display
+    assert one.decimal_places == 2, "default value"
+
+    two = Money(2, "EUR", decimal_places=decimal_places)
+
+    assert two._decimal_places_display is None, "default value"
+    assert two.decimal_places == decimal_places
+
+    three = Money(3, "EUR")
+    one._copy_attributes(two, three)
+
+    assert three._decimal_places_display == decimal_places_display
+    assert three.decimal_places == max(2, decimal_places) if decimal_places is not None else 2
+
+    zero = Money(0, "EUR")
+    one._copy_attributes(Money(1, "EUR", decimal_places_display=3), zero)
```

Comment:
Oh, I thought you'd keep arithmetic operations and replace `assert three._decimal_places_display == ...` with assertions on string formatting (as in the PR description)

Reply:
This is the function who has changed, by directly using this function in this test I had hoped to make the test more clear of what its testing purpose is. By using addition or subtraction it was masquerading it purpose.
The formatting randomness is a result of having the decimal_places_display incorrectly set and being used in selecting the format for str output.
So this fix is more about consistency then formatting, that is a nice extra result.

Reply:
Yep, makes sense